keps/2941-DRA-Structured-Parameters/README.md

```diff
@@ -0,0 +1,549 @@
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ResourceFlavor
+metadata:
+  name: "default-gpu-flavor"
+spec:
+  dynamicResources:
+  - name: whole-gpus
+    deviceClassNames:
+    - gpu.example.com                # 4.1.1.1.1) create a map{"gpu.example.com" -> "whole-gpus"}
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: Workload
+metadata:
+  name: job-job0-6f46e
+  namespace: gpu-test1
+...
+...
+status:
+  admission:
+    clusterQueue: single-gpus-cluster-queue
+    podSetAssignments:
+    - count: 2
+      flavors:
+        cpu: whole-gpu-flavor
+        memory: whole-gpu-flavor
+        single-gpus: default-gpu-flavor # 5) selected flavor is reflected here in workload status
+      name: main
+      resourceUsage:
+        cpu: "1"
+        memory: 400Mi
+        whole-gpus: "1"                # 5) selected device count is reflected here in workload status
+```
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+#### Unit Tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+TBD
+- `<package>`: `<date>` - `<test coverage>`
+
+#### Integration tests
+
+I am not sure if we can test DRA functionality (requiring alpha features enabled) at the integration level.
+
+DRA requires a kubelet plugin so this may not be a good candidate for an integration test.
+
+#### E2E Test
+
+Use existing dra-example-driver or Kubernetes test driver for e2e testing.
+
+### Graduation Criteria
+
+#### Feature Gate
+
+We will introduce a KueueDynamicResourceAllocation feature gate.
+
+This feature gate will go beta depending on community feedback of functionality in alpha.
+
+The goal will be limit changes only if this feature gate is enabled in combination with the DRA feature.
```

Comment:
We should explain when we can graduate this feature.

Reply:
@kannon92 can you please take a look at the Graduation Criteria, I took a stab at defining it, will like to get additional thoughts.